File: guides/micronaut-creating-first-graal-app/kotlin/src/main/kotlin/example/micronaut/ConferenceService.kt
package example.micronaut

import java.util.Random
import jakarta.inject.Singleton

@Singleton // <1>
class ConferenceService {

    fun randomConf(): Conference = CONFERENCES[Random().nextInt(CONFERENCES.size)] // <2>

    companion object {
        private val CONFERENCES = listOf(
                Conference("Greach"),
                Conference("GR8Conf EU"),
                Conference("Micronaut Summit"),
                Conference("Devoxx Belgium"),
                Conference("Oracle Code One"),
                Conference("CommitConf"),
                Conference("Codemotion Madrid")
        )
    }
}

File: guides/micronaut-creating-first-graal-app/python/src/example/micronaut/conference_service.py
import random

from jakarta.inject import Singleton

from .conference import Conference


@Singleton  # <1>
class ConferenceService:
    _conferences = [
        Conference("Greach"),
        Conference("GR8Conf EU"),
        Conference("Micronaut Summit"),
        Conference("Devoxx Belgium"),
        Conference("Oracle Code One"),
        Conference("CommitConf"),
        Conference("Codemotion Madrid"),
    ]

    def random_conf(self) -> Conference:  # <2>
        return random.choice(self._conferences)
